mintapi/api.py

```diff
@@ -331,38 +333,68 @@ def initiate_account_refresh(email, password):
+    import argparse
 
-    # Parse command-line arguments {{{
-    cmdline = optparse.OptionParser(usage = 'usage: %prog [options] [email password]')
-    cmdline.add_option('--accounts', action = 'store_true', dest = 'accounts', default = False, help = 'Retrieve account information (default if nothing else is specified)')
-    cmdline.add_option('--budgets', action = 'store_true', dest = 'budgets', default = False, help = 'Retrieve budget information')
-    cmdline.add_option('--extended-accounts', action = 'store_true', dest = 'accounts_ext', default = False, help = 'Retrieve extended account information (slower, implies --accounts)')
-    cmdline.add_option('--transactions', '-t', action='store_true', default=False, help='Retrieve transactions')
-    cmdline.add_option('--filename', '-f', help='write results to file. can be {csv,json} format. default is to write to stdout.')
-    cmdline.add_option('--user', '-u', help='mint email login. uses OS keyring to store password info.')
+    try:
+        import keyring
+    except ImportError:
+        keyring = None
 
-    (options, args) = cmdline.parse_args()
+    # Parse command-line arguments {{{
+    cmdline = argparse.ArgumentParser()
+    cmdline.add_argument('email', nargs='?', default=None,
+                         help='The e-mail address for your Mint.com account')
+    cmdline.add_argument('password', nargs='?', default=None,
+                         help='The password for your Mint.com account')
+    cmdline.add_argument('--accounts', action='store_true', dest='accounts',
+                         default=False, help='Retrieve account information'
+                         ' (default if nothing else is specified)')
+    cmdline.add_argument('--budgets', action='store_true', dest='budgets',
+                         default=False, help='Retrieve budget information')
+    cmdline.add_argument('--extended-accounts', action='store_true',
+                         dest='accounts_ext', default=False,
+                         help='Retrieve extended account information (slower, '
+                         'implies --accounts)')
+    cmdline.add_argument('--transactions', '-t', action='store_true',
+                         default=False, help='Retrieve transactions')
+    cmdline.add_argument('--filename', '-f', help='write results to file. can '
+                         'be {csv,json} format. default is to write to '
+                         'stdout.')
+    cmdline.add_argument('--keyring', action='store_true',
+                         help='Use OS keyring for storing password '
+                         'information')
+
+    options = cmdline.parse_args()
+
+    if options.keyring and not keyring:
+        cmdline.error('--keyring can only be used if the `keyring`'
+                      'library is installed.')
 
-    # Handle Python 3's raw_input change.
     try:
-        input = raw_input
+        from __builtin__ import raw_input as input
     except NameError:
         pass
```

Comment:
This code for fixing raw_input didn't really work. It caused an error for me on Python 3.
Apparently in Python 3, even trying to assign to input even if you fail, causes the built-in input to be blasted away.

The new code I added works in both Python 2 and Python 3 and is based on how the `six` compatibility library handles things.